testinfra/modules/service.py

```diff
@@ -113,6 +113,22 @@ def is_enabled(self):
         # https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=760616
         return super(SystemdService, self).is_enabled
 
+    @property
+    def is_valid(self):
+        # systemd-analyze requires a full path.
+        if self.name.endswith(".service"):
+            name = self.name
+        else:
+            name = self.name + ".service"
+        cmd = self.run("systemd-analyze verify %s", name)
+        # A bad unit file still returns a rc of 0, so check the
+        # stdout for anything.  Nothing means no warns/errors.
+        # Docs at https://www.freedesktop.org/software/systemd/man/systemd
+        # -analyze.html#Examples%20for%20verify
+        if not cmd.stdout and not cmd.stderr:
+            return True
+        return False
```

Comment:
Hi, thanks for contributing! And sorry for my late review.

This looks useful to testinfra and I learned how to verify a systemd service, thanks! :)
Nevertheless I'll do a small modification to have a better reporting because I guess users will use it with `assert service.is_valid` and we could provide more explanation by having stdout and stderr in the assertion reporting:

Maybe ```assert (cmd.stdout, cmd.stderr) == ("", "")``` ?

Reply:
No problem.

I thought of this when I was writing it but for some reason didn't do it.  Updated the PR.  Can rebase if you wish.